avm/res/synapse/workspace/firewall-rules/main.bicep

```
@@ -0,0 +1,37 @@
+metadata name = 'Synapse Workspaces Firewall Rules'
+metadata description = 'This module deploys Synapse Workspaces Firewall Rules.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Conditional. The name of the parent Synapse Workspace. Required if the template is used in a standalone deployment.')
+param workspaceName string
+
+@description('Required. The name of the firewall rule.')
+param name string
+
+@description('Required. The start IP address of the firewall rule. Must be IPv4 format.')
+param startIpAddress string
+
+@description('Required. The end IP address of the firewall rule. Must be IPv4 format. Must be greater than or equal to startIpAddress.')
+param endIpAddress string
+
+resource workspace 'Microsoft.Synapse/workspaces@2021-06-01' existing = {
+  name: workspaceName
+}
+
+resource firewallrules 'Microsoft.Synapse/workspaces/firewallRules@2021-06-01' = {
```

```suggestion
resource firewallRule 'Microsoft.Synapse/workspaces/firewallRules@2021-06-01' = {
```
I think this should be singular and camelCase (as per [specs](https://azure.github.io/Azure-Verified-Modules/specs/bicep/#id-bcpnfr8---category-composition---code-styling---lower-camelcasing))

---

Un-resolved as not addressed

---

updated